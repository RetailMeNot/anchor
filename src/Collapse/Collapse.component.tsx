// REACT
import * as React from 'react';
// VENDOR
import classNames from 'classnames';
import styled, { css } from '@xstyled/styled-components';
import { variant as createVariant, th } from '@xstyled/system';
// Components
import { ChevronDown, ChevronUp } from '../Icon';

export type CollapseVariants = 'comfortable' | 'compact' | 'none' | string;

interface CollapseProps extends React.HTMLAttributes<HTMLDivElement> {
    /**
     * Whether the component starts off open or closed. Can also be used to programmatically open/close the
     * component.
     */
    isOpen?: boolean;
    /** Text to show for the opened state of the component */
    openedText?: string;
    /** Text to show for the closed state of the component */
    closedText?: string;
    /** Icon to show for the opened state of the component */
    openedIcon?: React.ReactElement;
    /** Icon to show for the closed state of the component */
    closedIcon?: React.ReactElement;
    /** 2 possible variants, compact (doc site) and comfortable (RMN CBO), and also none to allow for styling */
    variant?: CollapseVariants;
    /** This prop is solely used by the CollapseGroup component to make Collapse components have accordion behavior */
    onClick?: any;
    /** This mostly exists for CollapseGroup in order to hide the bottom border of stacked Collapse children */
    hasBottomBorder?: boolean;
    children?: any;
    className?: string;
}

interface StyledCollapseProps extends CollapseProps {
    variant?: string;
    hasBottomBorder?: boolean;
}

export const DEFAULT_OPENED_TEXT = 'Close';
export const DEFAULT_CLOSED_TEXT = 'Open';
const DEFAULT_VARIANT = 'comfortable';

const variants = {
    comfortable: css`
        padding: 1rem 2rem;
        border-top: solid thin ${th.color('borders.base')};
        border-bottom: solid thin ${th.color('borders.base')};

        .anchor-collapse-button {
            cursor: pointer;
            display: block;
            width: 100%;
            text-align: left;
            border-style: none;
            font-weight: 500;
            font-size: 0.875rem;
            padding: 0.5rem 0;
            user-select: none;

            &:focus {
                outline: none;
            }

            span:last-child {
                float: right;
            }
        }

        .anchor-collapse-content {
            font-size: 1rem;
            text-align: left;
            padding: 0.5rem 0;
        }
    `,
    compact: css`
        color: neutrals.charcoal.light;

        .anchor-collapse-button {
            background-color: neutrals.white.base;
            border-style: none;
            border-bottom: solid thin ${th.color('borders.light')};
            border-top: solid thin ${th.color('borders.light')};
            color: neutrals.charcoal.light;
            cursor: pointer;
            display: block;
            font-weight: bold;
            height: 3rem;
            padding: 0 1.3125rem;
            text-align: left;
            user-select: none;
            width: 100%;

            &:focus {
                outline: none;
            }

            *:last-child {
                float: right;
            }
        }

        .anchor-collapse-content {
            background-color: neutrals.silver.light;

            ul {
                padding: 0;
                margin: 0;
                list-style-type: none;
            }
            // TODO: we shouldn't be using as many fragile nested selectors and we should be using Typography
            li a {
                display: block;
                padding-left: 3.4375rem;
                height: 2.6875rem;
                line-height: 2.6875rem;
                cursor: pointer;
                font-size: 0.875rem;
                color: #222;

                &:hover {
                    background-color: rgba(0, 126, 205, 0.1);
                }
            }
        }
    `,
    none: css({}),
};

const variantStyles = createVariant({
    key: 'collapse.variants',
    prop: 'variant',
    default: DEFAULT_VARIANT,
    variants: variants,
});

const StyledCollapse = styled('div')<StyledCollapseProps>`
    display: block;
    box-sizing: border-box;
    font-family: base;

    ${variantStyles}

    &.no-bottom-border {
        border-bottom-style: none;

        .anchor-collapse-button {
            border-bottom-style: none;
        }
    }
`;
StyledCollapse.displayName = 'StyledCollapse';

export const Collapse: React.FunctionComponent<CollapseProps> = ({
    isOpen = false,
    openedText = DEFAULT_OPENED_TEXT,
    closedText = DEFAULT_CLOSED_TEXT,
    openedIcon,
    closedIcon,
    variant = DEFAULT_VARIANT,
    onClick,
    hasBottomBorder = true,
    className,
    children,
    ...props
}: CollapseProps): React.ReactElement<CollapseProps> => {
    const [open, toggleOpen] = React.useState<boolean>(isOpen);
    const IconOpened = openedIcon || <ChevronDown />;
    const IconClosed = closedIcon || <ChevronUp />;

    // If the user provides custom openedText but not custom closedText
    // then use openedText for closedText as well
    const textClosed =
        openedText !== DEFAULT_OPENED_TEXT && closedText === DEFAULT_CLOSED_TEXT
            ? openedText
            : closedText;

    // Allows for the programmatic opening/closing of the Collapse
    // without a user having to click it
    React.useEffect(() => {
        toggleOpen(isOpen);
    }, [isOpen]);

    return (
        <StyledCollapse
            variant={variant}
            className={classNames(
                'anchor-collapse',
                open && 'open',
                variant,
                !hasBottomBorder && 'no-bottom-border',
                className
            )}
            {...props}
        >
            <button
                onClick={onClick || (() => toggleOpen(!open))}
                className="anchor-collapse-button"
            >
                {open ? (
                    <React.Fragment>
                        {openedText} {IconOpened}
                    </React.Fragment>
                ) : (
                    <React.Fragment>
                        {textClosed} {IconClosed}
                    </React.Fragment>
                )}
            </button>

            {open && (
                <section className="anchor-collapse-content">
                    {children}
                </section>
            )}
        </StyledCollapse>
    );
};
