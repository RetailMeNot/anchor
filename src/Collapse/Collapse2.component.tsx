// VENDOR
import * as React from 'react';
import styled, { css } from '@xstyled/styled-components';
import { space as spaceStyles, SpaceProps } from '@xstyled/system';
// ANCHOR
import { Button, ChevronDown, ChevronUp } from '../';
// UTILS
import { get } from '../utils/get/get';

enum enumIconAlignment {
    left = 'left',
    right = 'right',
}

type IconAlignment = keyof typeof enumIconAlignment;

interface Collapse2Props {
    children?: any;
    closedIcon?: React.ReactElement;
    closedText?: string;
    iconAlignment?: IconAlignment;
    openedIcon?: React.ReactElement;
    openedText?: string;
}

/* Main wrapper */
const StyledCollapse = styled('div')`
    border-top: base;
    border-bottom: base;
`;

/* Extending Button to expose some styling options */
const CollapseButton = styled(Button)`
    color: text.base;
    font-weight: normal;
    ${({prefix}) => css({ justifyContent: prefix ? 'end' : 'space-between'})}
`;

interface StyledOverflowContainer {
    isOpen: boolean;
    openHeight: string;
}

/* An overflow container that hides children when collapsed */
const StyledOverflowContainer = styled('div')<StyledOverflowContainer>`
    overflow-y: hidden;
    transition: all 1000ms;
    ${({isOpen, openHeight}) =>  css({ height: isOpen ? openHeight + '!important' : '0px !important'})}
`;

/* Container of the children. This is what is used to get the height of the content. */
const StyledContainer = styled('div')<SpaceProps>`
    ${spaceStyles}
`;

export const Collapse2 = ({
    children,
    closedIcon = <ChevronUp />,
    openedIcon = <ChevronDown />,
    closedText = 'Click to open',
    openedText = 'Click to close',
    iconAlignment = enumIconAlignment.right,
}: Collapse2Props) => {
    const container = React.useRef<HTMLDivElement>(null);
    const overflowContainer = React.useRef<HTMLDivElement>(null);
    const [isOpen, setIsOpen] = React.useState<boolean>(false);
    const [openHeight, setOpenHeight] = React.useState<string>('');
    const buttonText = isOpen ? openedText : closedText;

    /*
        Once the component is mounted the height of the container is recorded and made an inline-style.
        This is necessary in order to have the sliding animation of showing/hiding content.
    */
    React.useEffect(() => {
        const containerHeight = get(container.current, 'clientHeight', 0);

        if (container && container.current && overflowContainer && overflowContainer.current) {
            overflowContainer.current.style.height = containerHeight + 'px';
            setOpenHeight(containerHeight);
        }
    }, []);

    return(
        <StyledCollapse>
            <CollapseButton
                block
                onClick={() => setIsOpen(!isOpen)}
                variant="minimal"
                prefix={
                    iconAlignment === enumIconAlignment.left
                        ? isOpen
                        ? closedIcon
                        : openedIcon
                        : undefined
                }
                suffix={
                    iconAlignment === enumIconAlignment.right
                        ? isOpen
                        ? closedIcon
                        : openedIcon
                        : undefined
                }
            >
                {buttonText}
            </CollapseButton>

            <StyledOverflowContainer
                isOpen={isOpen}
                openHeight={openHeight}
                ref={overflowContainer}
            >
                <StyledContainer ref={container} p="2">
                    {children}
                </StyledContainer>
            </StyledOverflowContainer>
        </StyledCollapse>
    );
};
