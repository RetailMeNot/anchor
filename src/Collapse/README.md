# Collapse

The Collapse component allows a user to click on a button and toggle the rendering of its child.

## Usage

    import { Collapse } from '@retailmenot/anchor';

    const example = () => (
        <Collapse>
            I will appear and disappear when the rendered button is clicked!
        </Collapse>
    );

## Settings/Props

The component can accept the following `props`:

* `isOpen` *(boolean)*: Defaults to **`false`**. Determines whether the Collapse component is open or not. Changing the value of this prop will programatticaly open/close the component, thusly not requiring a click event.

* `openText` *(string)*: Defaults to **`'Open'`**. The text that appears on the button as a Call To Action (CTA) to open the collapsed content. If this prop is used and the `closeText` prop is not, `closeText` will use whatever is defined for `openText`. This is useful if you don't actually want the CTA text to change.

* `closeText` *(string)*: Defaults to **`'Close'`**. The text that appears on the button as a CTA to close the displayed content.

* `openIcon` *(React.ReactElement)*: Defaults to **`<ChevronDown />`** from `Icon`. Accepts any component to render, although realistically sticking to components from `Icon` make the most sense.

* `closeIcon` *(React.ReactElement)*: Defaults to **`<ChevronUp />`** from `Icon`. Accepts any component to render, although realistically sticking to components from `Icon` make the most sense.

    #### `openIcon` and/or `closeIcon` usage

        <Collapse
            openIcon={<ArrowLeft />}
            closeIcon={<ArrowRight />}
        >
            <div>Hello world.</div>
        </Collapse>

* `theme` *('comfortable' | 'compact' | 'none')*: Defaults to **`'comfortable'`**.
    * **comfortable**: has considerable spacing. It is intended to be used for showing/hiding larger pieces of content.
    * **compact**: has much tighter spacing. It is intended to be used for showing/hiding navigation elements.
    * **none**: Applies no styling whatsoever to the component. Useful for when you want complete control of all styling via styled-components/css/etc., but still keep the functionality of the component. The component has many classes in place to try and make it easier to accomplish this.

* `hasBottomBorder`: *(boolean)*: Defaults to **`true`**. Shows/hides the bottom border.

## Dev Notes

**`Collapse`** does have an additional prop, **`toggleFn`** , however it is strongly suggested not to use it. Its sole reason to exist is to allow the **`CollapseGroup`** component to take control of the **`isOpen`** prop. Also of note is the **`hasBottomBorder`** prop.

* `toggleFn`: Whatever function is passed to this prop will be executed when the user clicks the **`.collapse-button`** element. It will not intrinsically toggle the **`isOpen`** prop.

* `hasBottomBorder`: This mostly exists to allow **`CollapseGroup`** to control whether or not a **`Collapse`** child component should show its bottom border. All bottom borders are hidden save for the last child in order to make it look like one unified experience.
<br /><br />

# CollapseGroup

The **`CollapseGroup`** component combines multiple **`Collapse`** components together, allowing a single component to apply props to all children, unifying their appearance, and adding the ability to have accordion functionality.

## Usage

    import { Collapse, CollapseGroup } from '@retailmenot/anchor';

    const example = () => (
        <CollapseGroup theme="compact">
            <Collapse>
                I will appear and disappear when the rendered button is clicked!
            </Collapse>
            <Collapse>
                I will also appear and disappear when the rendered button is clicked!
            </Collapse>
        </CollapseGroup>
    );

## Settings/Props

The component can accept the following `props`:

* `openIcon` *(React.ReactElement)*: No default. Using this prop will apply the setting to all child **`Collapse`** components. Otherwise its functionality is as noted above for **`Collapse`**.

* `closeIcon` *(React.ReactElement)*: No default. Using this prop will apply the setting to all child **`Collapse`** components. Otherwise its functionality is as noted above for **`Collapse`**.

* `theme` *('comfortable' | 'compact' | 'none')*: No default. Using this prop will apply the setting to all child **`Collapse`** components. Otherwise its functionality is as noted above for **`Collapse`**.

* `accordion` *(boolean)*: Defaults to **`false`**. Enables accordion functionality, where only one **`Collapse`** component can be open at a time.

* `openIndex` *(number)*: Defaults to **`0`**. If **`accordion`** is set to **`true`**, this determines which **`Collapse`** component will start in the open state.

## Dev Notes

**`openIndex`** does not have a knob because Storybook does not correctly update the component in realtime. Explicitly setting the value does work, however.