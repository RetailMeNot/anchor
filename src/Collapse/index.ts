export { Collapse } from './Collapse.component';
