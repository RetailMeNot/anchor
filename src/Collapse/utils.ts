// VENDOR
import styled from 'styled-components';

type ScaleFactors = 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'xxl';

export interface IconSVGProps {
    scale?: ScaleFactors;
    color?: string;
    className?: string;
    children?: any;
}

export const Scale = {
    xs: 8,
    sm: 14,
    md: 16,
    lg: 24,
    xl: 32,
    xxl: 48,
};

export const DefaultColor = 'currentColor';
export const DefaultScale = 'md';

export const StyledIcon = styled.span<IconSVGProps>`
    display: inline-block;
    height: ${({ scale = 'md' }: IconSVGProps) => `${Scale[scale] / 16}rem`};
    width: ${({ scale = 'md' }: IconSVGProps) => `${Scale[scale] / 16}rem`};
    line-height: 0;
`;

StyledIcon.displayName = 'AnchorIcon';
