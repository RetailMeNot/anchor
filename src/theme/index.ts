export { default as colors } from './colors.theme';
export { default as fonts } from './fonts.theme';
export { default as sizes } from './sizes.theme';
export { default as variables } from './variables.theme';

export { NormalizeCSS, GlobalCSS } from './GlobalStyles';
