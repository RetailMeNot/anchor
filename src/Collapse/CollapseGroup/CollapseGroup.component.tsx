// REACT
import * as React from 'react';
// VENDOR
import classNames from 'classnames';
import styled from 'styled-components';
// COMPONENTS
import { Themes } from '../Collapse.component';

// Currently no styles but following the design pattern of anchor components where top-level components
// are styled components.
const StyledCollapseGroup = styled.div``;
StyledCollapseGroup.displayName = 'StyledCollapseGroup';

interface CollapseGroupProps {
    /** Whether or not the CollapseGroup acts as an accordion, i.e. only one Collapse child can be open at a time. */
    accordion?: boolean;
    /** A zero-based index to indicate which Collapse child should be open first. Default is 0. */
    openIndex?: number;
    /** By passing a theme to the group, all Collapse children take on that theme */
    theme?: Themes;
    /** By passing an openIcon to the group, all Collapse children use that openIcon */
    openIcon?: React.ReactElement;
    /** By passing a closeIcon to the group, all Collapse children use that closeIcon */
    closeIcon?: React.ReactElement;
    // tslint:disable-next-line: max-line-length
    /** Realistically anything can be a child of CollapseGroup, but it's rather pointless as it's intended to have Collapse components as children */
    children?: any;
    className?: string;
}

export const CollapseGroup = ({
    accordion = false,
    openIndex = 0,
    theme,
    openIcon,
    closeIcon,
    children,
    className,
}: CollapseGroupProps) => {
    const [_openIndex, setOpenIndex] = React.useState<number>(openIndex);

    return (
        <StyledCollapseGroup
            className={classNames('anchor-collapse-group', className)}
        >
            {
                // tslint:disable-next-line: max-line-length
                /** If accordion is true, all Collapse components are cloned in order to control their isOpen prop vis accordionHandler.
                 *  Additionally, they are cloned to apply global settings, like theme, to all Collapse children.
                 */
            }
            {children &&
                  children.map((child: any, i: number) => {
                      return React.cloneElement(child, {
                          isOpen: accordion ? i === _openIndex : undefined,

                          // If accordion, use setOpenIndex as the event handler to control accordion functionality
                          // of the Collapse child. If the _openIndex is the same as the iterator (i) that means
                          // the Collapse element is already open, in which case _openIndex gets set to -1 which ends up
                          // returning false for all checks allowing an opened Collapse element to be closed.
                          accordionHandler: accordion ? () => setOpenIndex(i === _openIndex ? -1 : i) : undefined,

                          theme: theme,
                          openIcon: openIcon,
                          closeIcon: closeIcon,

                          // No Collapse element should have a bottom border except for the last one.
                          hasBottomBorder: i === children.length - 1,

                          key: `collapse-child-${i}`,
                      });
                  })
            }
        </StyledCollapseGroup>
    );
};
