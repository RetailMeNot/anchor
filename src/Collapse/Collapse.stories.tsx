// REACT
import * as React from 'react';
// STORYBOOK
import { storiesOf } from '@storybook/react';
import { text, select, boolean } from '@storybook/addon-knobs';
// VENDOR
import styled from 'styled-components';
// COMPONENTS
import {
    Collapse,
    DEFAULT_OPEN_TEXT,
    DEFAULT_CLOSE_TEXT,
} from './Collapse.component';
import * as Icon from '../Icon';
import { colors } from '../theme';

// README
import * as README from './README.md';

const TEXT = 'text';

const StyledComfortableStory = styled.div`
    padding: 2rem 5rem;
`;

storiesOf('Components/Collapse', module)
    .addParameters({
        readme: {
            sidebar: README,
        },
    })
    .add('Default', () => {
        const iconOpenPick = select(
            'Example Icon Open',
            Object.keys(Icon),
            'ChevronUp'
        );
        const iconClosePick = select(
            'Example Icon Close',
            Object.keys(Icon),
            'ChevronDown'
        );
        const content = select('Example Content', [TEXT, 'list'], TEXT);

        return (
            <StyledComfortableStory>
                <Collapse
                    isOpen={boolean('isOpen', false)}
                    openText={text('openText', DEFAULT_OPEN_TEXT)}
                    closeText={text('closeText', DEFAULT_CLOSE_TEXT)}
                    theme={select(
                        'theme',
                        ['compact', 'comfortable', 'none'],
                        'comfortable'
                    )}
                    openIcon={React.createElement(Icon[iconOpenPick], {
                        color: colors.savvyCyan.base,
                    })}
                    closeIcon={React.createElement(Icon[iconClosePick], {
                        color: colors.savvyCyan.base,
                    })}
                >
                    {content === TEXT ? (
                        <div>
                            Lorem ipsum dolor sit amet, consectetur adipiscing
                            elit. Nam eros ante, lacinia at imperdiet
                            consectetur, ultrices in elit. Vivamus vehicula nisl
                            id congue pulvinar.
                        </div>
                    ) : (
                        <ul>
                            <li>
                                <a>Grids & Layout</a>
                            </li>
                            <li>
                                <a>Cards</a>
                            </li>
                            <li>
                                <a>Color Usage</a>
                            </li>
                            <li>
                                <a>Typography</a>
                            </li>
                            <li>
                                <a>Icons</a>
                            </li>
                            <li>
                                <a>Animation</a>
                            </li>
                        </ul>
                    )}
                </Collapse>
            </StyledComfortableStory>
        );
    });
