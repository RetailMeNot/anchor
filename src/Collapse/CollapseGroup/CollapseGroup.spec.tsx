// REACT
import * as React from 'react';
import * as renderer from 'react-test-renderer';
// VENDOR
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from '@xstyled/styled-components';
// COMPONENT
import { RootTheme } from '../../theme';
import { CollapseGroup } from './CollapseGroup.component';
import { Collapse } from '../Collapse.component';

describe('Component: CollapseGroup', () => {
    it('should match its snapshot.', () => {
        const subject = (
            <ThemeProvider theme={RootTheme}>
                <CollapseGroup variant="comfortable">
                    <Collapse>Hello World</Collapse>
                    <Collapse>Hello World</Collapse>
                    <Collapse>Hello World</Collapse>
                </CollapseGroup>
            </ThemeProvider>
        );
        const component = shallow(subject);

        expect(subject).toBeDefined();
        expect(component).toBeDefined();

        const tree = renderer.create(subject).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it("should show only one Collapse component's content when accordion is true.", () => {
        const subject = (
            <ThemeProvider theme={RootTheme}>
                <CollapseGroup accordion openIndex={1}>
                    <Collapse>Hello World</Collapse>
                    <Collapse>Hello World</Collapse>
                    <Collapse>Hello World</Collapse>
                </CollapseGroup>
            </ThemeProvider>
        );
        const wrapper = mount(subject);
        const toggleButton = wrapper.find('.anchor-collapse-button').first();
        toggleButton.simulate('click');

        expect(
            wrapper.find('StyledCollapse.anchor-collapse.open')
        ).toHaveLength(1);
    });

    // tslint:disable-next-line: max-line-length
    it("should assign props specified on the component to child Collapse child component's, and override them.", () => {
        const subject = (
            <ThemeProvider theme={RootTheme}>
                <CollapseGroup variant="comfortable">
                    <Collapse>Hello World</Collapse>
                    <Collapse variant="none">Hello World</Collapse>
                    <Collapse>Hello World</Collapse>
                </CollapseGroup>
            </ThemeProvider>
        );
        const component = mount(subject);

        expect(
            component.find('StyledCollapse.anchor-collapse.comfortable')
        ).toHaveLength(3);
    });

    it('should remove the bottom border of Collapse child components.', () => {
        const subject = (
            <ThemeProvider theme={RootTheme}>
                <CollapseGroup>
                    <Collapse>Hello World</Collapse>
                    <Collapse>Hello World</Collapse>
                    <Collapse>Hello World</Collapse>
                </CollapseGroup>
            </ThemeProvider>
        );
        const wrapper = mount(subject);

        expect(
            wrapper
                .find('StyledCollapse.anchor-collapse')
                .at(0)
                .hasClass('no-bottom-border')
        ).toEqual(true);
        expect(
            wrapper
                .find('StyledCollapse.anchor-collapse')
                .at(1)
                .hasClass('no-bottom-border')
        ).toEqual(true);
        expect(
            wrapper
                .find('StyledCollapse.anchor-collapse')
                .at(2)
                .hasClass('no-bottom-border')
        ).toEqual(false);
    });
});
