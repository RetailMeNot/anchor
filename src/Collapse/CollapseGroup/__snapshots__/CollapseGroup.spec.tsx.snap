// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CollapseGroup should match its snapshot. 1`] = `
.c1 {
  display: inline-block;
  height: 1rem;
  width: 1rem;
  line-height: 0;
  color: currentColor;
}

.c0 {
  display: block;
  box-sizing: border-box;
  font-family: Avenir Next,Segoe UI,Helvetica Neue,Helvetica,Roboto,sans-serif;
  padding: 1rem 2rem;
  border-top: solid thin #D3D3D3;
  border-bottom: solid thin #D3D3D3;
}

.c0 .anchor-collapse-button {
  cursor: pointer;
  display: block;
  width: 100%;
  text-align: left;
  border-style: none;
  font-weight: 500;
  font-size: 0.875rem;
  padding: 0.5rem 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 .anchor-collapse-button:focus {
  outline: none;
}

.c0 .anchor-collapse-button span:last-child {
  float: right;
}

.c0 .anchor-collapse-content {
  font-size: 1rem;
  text-align: left;
  padding: 0.5rem 0;
}

.c0.no-bottom-border {
  border-bottom-style: none;
}

.c0.no-bottom-border .anchor-collapse-button {
  border-bottom-style: none;
}

<div
  className="anchor-collapse-group"
>
  <div
    className="c0 anchor-collapse comfortable no-bottom-border"
  >
    <button
      className="anchor-collapse-button"
      onClick={[Function]}
    >
      Open
       
      <span
        className="c1 anchor-icon chevron-up"
        scale="md"
      >
        <svg
          height={16}
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 7.523l2.954 2.518a.77.77 0 0 0 1.058-.001l.014-.014a.721.721 0 0 0-.009-1.052L8.53 5.96a.767.767 0 0 0-1.048-.009l-3.506 3.03a.719.719 0 0 0-.001 1.043l.015.015a.767.767 0 0 0 1.048.01L8 7.523z"
            fill="currentColor"
            fillRule="nonzero"
          />
        </svg>
      </span>
    </button>
  </div>
  <div
    className="c0 anchor-collapse comfortable no-bottom-border"
  >
    <button
      className="anchor-collapse-button"
      onClick={[Function]}
    >
      Open
       
      <span
        className="c1 anchor-icon chevron-up"
        scale="md"
      >
        <svg
          height={16}
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 7.523l2.954 2.518a.77.77 0 0 0 1.058-.001l.014-.014a.721.721 0 0 0-.009-1.052L8.53 5.96a.767.767 0 0 0-1.048-.009l-3.506 3.03a.719.719 0 0 0-.001 1.043l.015.015a.767.767 0 0 0 1.048.01L8 7.523z"
            fill="currentColor"
            fillRule="nonzero"
          />
        </svg>
      </span>
    </button>
  </div>
  <div
    className="c0 anchor-collapse comfortable"
  >
    <button
      className="anchor-collapse-button"
      onClick={[Function]}
    >
      Open
       
      <span
        className="c1 anchor-icon chevron-up"
        scale="md"
      >
        <svg
          height={16}
          viewBox="0 0 16 16"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 7.523l2.954 2.518a.77.77 0 0 0 1.058-.001l.014-.014a.721.721 0 0 0-.009-1.052L8.53 5.96a.767.767 0 0 0-1.048-.009l-3.506 3.03a.719.719 0 0 0-.001 1.043l.015.015a.767.767 0 0 0 1.048.01L8 7.523z"
            fill="currentColor"
            fillRule="nonzero"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
